libsolidity/codegen/ABIFunctions.h

```diff
@@ -66,6 +66,16 @@ class ABIFunctions
 		bool _encodeAsLibraryTypes = false
 	);
 
+	/// @returns name of an assembly function to ABI-decode values of @a _types
+	/// into memory. If @a _fromMemory is true, decodes from memory instead of
+	/// from calldata.
+	/// Can allocate memory.
+	/// Inputs: <source_offset> <source_end> (layout reversed on stack)
+	/// Outputs: <value0> <value1> ... <valuen>
+	/// The values represent stack slots. If a type occupies more or less than one
+	/// stack slot, it takes exactly that number of values.
+	std::string tupleDecoder(TypePointers const& _types, bool _fromMemory = false);
```

Comment:
So it returns the function name which has a signature of `<name>(source_offset, source_end) -> <v0>, <v1>, ..., <vn>`. is that correct?

Reply:
yes

Reply:
Cool, I was going to try it in the IR pull request and that API should work well.